source/common/upstream/health_discovery_service.cc

```diff
@@ -124,6 +162,9 @@ envoy::service::health::v3::HealthCheckRequestOrEndpointHealthResponse HdsDelega
             NOT_REACHED_GCOVR_EXCL_LINE;
           }
         }
+
+        // copy this endpoint's health info to the legacy flat-list.
```

Comment:
Should there be a TODO to remove this when the API advances to v4?

Reply:
I think this is a good idea. Transition is easy, to just remove this one line. I went ahead and added the comment.